### PR TITLE
B2C-6912-reduce-to-BE

### PR DESCRIPTION
**Pull request by rkorzeniec**
@amrdruid and myself were asked to simplify the FE task to facilitate BE engineer hiring.

This PR:
- reorganises the file structure
- moves FE challenge from readme to own file
- adds BE (modified FE) challenge
- updates readme

### CODE REVIEW

**Comment by rkorzeniec** [comment]
@severin @juliabecker just an FYI if you still wish to be in the loop

File: backend.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+- `addresses[0].longitude`
+
+_Note:_ The data is completely random and does not "make sense", so it's perfectly possible that the coordinates mark a place in Europe while the country reads "Taiwan". Do not freak out about this :)
+
+## Implementation Notes
+
+You're free to use any Ruby framework of your choice, whether it's Rails, Sinatra, or Grape ... or no framework at all. Also good to keep in mind:
+
+- Sufficient test coverage
+- We expect you to be prepared to explain your rationale behind the decisions taken. We also expect you to provide us with your assumptions. This will help us contextualise your solution when assessing the task.
```
Comment (amrdruid) [inline backend.md]: contextualize

**Review by amrdruid — APPROVED** [review]
Grammarly is happy just one small thing

Ready to go! Thanks so much Robert for that work

**Comment by amrdruid** [comment]
Actually, one small thing, do we need to mention `dealers_controller#index` in the readme?

File: backend.md (hunk 1 above)
Comment (rkorzeniec) [reply backend.md]: clearly gramarly is american, not british

Comment (rkorzeniec) [reply backend.md:46]: long story short both are correct, happy to change but with `contextualise` we are more in line as we have `synchronise` etc.

**Comment by rkorzeniec** [comment]
> Actually, one small thing, do we need to mention `dealers_controller#index` in the readme?

I would not, it is an implementation detail and so far we have not provided such. I would not do it IMO, what if they want to name it differently? Naming things well is an important skill and we do mention `Provide an API endpoint to return the dealer data` already. My 2 cents on it

**Review by amrdruid — APPROVED** [review]
🌟

**Pull request merged**